README.md

```diff
@@ -26,4 +27,47 @@ Follow instructions here https://cloud.google.com/cloud-build/docs/configure-thi
 | GCLOUD_PROJECT_ID                     | no default                             | GCP project id                                                                                       |
 | SLACK_WEBHOOK_URL                     | no default                             | Slack Webhook URL. Read more https://api.slack.com/incoming-webhooks                                 |
 | GCLOUD_PUBSUB_SUBSCRIPTION_NAME       | cloudbuild-notifier-subscription       | Google Cloud Pub/Sub topic subscription. Read more: https://cloud.google.com/pubsub/docs/subscriber  |
+| NOTIFICATION_FILTERS                     | []                             | A list of filters that should trigger notifications to be sent to Slack (see section below) By default, notifications will be sent for all build messages |
 
+##### NOTIFICATION_FILTERS (Optional)
+The NOTIFICATION_FILTERS environment variable should be passed as a JSON string consisting of a list of objects as in the case below:
```

Comment:
This seems like it should work, but I'm not super excited about having to pass in a JSON string. It seems a lot more complex than it needs to be.  I'm also not sure that we need to pass in an operator since the list of filters is essentially an OR filter (in other words, you can just pass in each version as another member of the list).

Can we do something like:

`<source regex>:<branch regex>:<status regex>,<source regex>:<branch regex>:<status regex>`

So then I can pass in something like:

`org-name/repo-name-(1,2):(main|production|dev-*):(SUCCESS|WORKING),org-name/repo-name-(3,4):(dev):(FAILURE|QUEUED)`

Implementation should be fairly straight forward as well - you just need to split the filters by `,` then for each filter, split into 3 regexes by `:`, then apply each regex.  All three regexes would need to pass (essentially an AND) for the filter to work.

Reply:
I agree that the use of regexes is a more elegant solution that's easier for users.
I have modified the notifier to use regex strings instead of JSON strings

Reply:
LGTM